searchlib/src/vespa/searchlib/tensor/hnsw_index_base.cpp

```diff
@@ -65,18 +64,16 @@ HnswIndexBase::get_link_array(uint32_t docid, uint32_t level) const
 {
     auto levels = get_level_array(docid);
     assert(level < levels.size());
-    return _links.get(levels[level]);
+    return _links.get(levels[level].load_acquire());
 }
 
 void
 HnswIndexBase::set_link_array(uint32_t docid, uint32_t level, const LinkArrayRef& links)
 {
     auto links_ref = _links.add(links);
-    // TODO: Add memory barrier?
-    auto node_ref = _node_refs[docid];
+    auto node_ref = _node_refs[docid].load_acquire();
```

Comment:
If this is always invoked from the writer thread, it's technically sufficient to use a `load_relaxed()` here since the changes to these structures shall always be visible to that thread anyway. But acquire loads are almost free on x86-64 anyway, so it's hardly something to prematurely optimize for.